Classes/common/touchdb/TDBlobStore.m

```diff
@@ -187,6 +190,8 @@ - (NSInteger)deleteBlobsExceptWithKeys:(NSSet*)keysToKeep withDatabase:(FMDataba
             [NSData dataWithBytes:oneRow.key.bytes length:sizeof(oneRow.key.bytes)];
         if ([keysToKeep containsObject:curKeyData]) {
             // Do not delete blob. It is an exception.
+            [filesToKeep addObject:oneRow.blobFilename];
+
             continue;
         }
 
```

Comment:
But if I do that, I will not update the value of [numDeleted](https://github.com/cloudant/CDTDatastore/pull/170/files#diff-ce77b2812a143c42aa7fdb8b3818581dR207) ... Uhm I realise of another situation:
1. The file is not deleted so I do not remove the row from db
2. I tried to delete again the same file in `deleteFilesExceptWithFilenames:` and this time, it works
I'll end up with a row in the database without a file.